skimage/measure/segmentation_metrics.py

```diff
@@ -0,0 +1,493 @@
+    """Compute Rand and Adjusted Rand indices for each threshold of a UCM
+
+    Parameters
+    ----------
+    im_true : ndarray
+        Ground-truth image.
+    im_test : ndarray
+        Test image.
+    npoints : int, optional
+        If provided, only compute values at npoints thresholds, rather than
+        all thresholds. Useful when ucm has an extremely large number of
+        unique values.
+
+    Returns
+    -------
+    ris : ndarray of float, shape (3, len(np.unique(ucm))) or (3, npoints)
+        The rand indices of the segmentation induced by thresholding and
+        labeling `ucm` at different values. The 3 rows of `ris` are the values
+        used for thresholding, the corresponding Rand Index at that threshold,
+        and the corresponding Adjusted Rand Index at that threshold.
+    """
+    _assert_compatible(im_true, im_test)
+
+    ts = np.unique(im_test)[1:]
+    if npoints is None:
+        npoints = len(ts)
+    if len(ts) > 2 * npoints:
+        ts = ts[np.arange(1, len(ts), len(ts) / npoints)]
+    result = np.zeros((2, len(ts)))
+    for i, t in enumerate(ts):
+        seg = label(im_test < t)[0]
+        result[0, i] = rand_index(seg, im_true)
+        result[1, i] = adj_rand_index(seg, im_true)
+    return np.concatenate((ts[np.newaxis, :], result), axis=0)
+
+def compare_vi(im1, im2, weights=np.ones(2)):
+    """Return the variation of information metric. [1]
+
+    VI(X, Y) = H(X | Y) + H(Y | X), where H(.|.) denotes the conditional
+    entropy.
+
+    Parameters
+    ----------
+    im1, im2 : ndarray
+        Image.  Any dimensionality.
+    weights : ndarray of float, shape (2,), optional
+        The weights of the conditional entropies of `im1` and `im2`. Equal weights
+        are the default.
+
+    Returns
+    -------
+    v : float
+        The variation of information between `im1` and `im2`.
+
+    References
+    ----------
+    [1] Meila, M. (2007). Comparing clusterings - an information based
+    distance. Journal of Multivariate Analysis 98, 873-895.
+    """
+    return np.dot(weights, compare_split_vi(im1, im2))
+
+def compare_split_vi(im1, im2):
+    """Return the symmetric conditional entropies associated with the VI.
+
+    The variation of information is defined as VI(X,Y) = H(X|Y) + H(Y|X).
+    If Y is the ground-truth segmentation, then H(Y|X) can be interpreted
+    as the amount of under-segmentation of Y and H(X|Y) is then the amount
+    of over-segmentation.  In other words, a perfect over-segmentation
+    will have H(Y|X)=0 and a perfect under-segmentation will have H(X|Y)=0.
+
+    Parameters
+    ----------
+    im1, im2 : ndarray
+        Image.  Any dimensionality.
+
+    Returns
+    -------
+    sv : ndarray of float, shape (2,)
+        The conditional entropies of im2|im1 and im1|im2.
+
+    See Also
+    --------
+    vi
+    """
+    _, _, _ , hxgy, hygx, _, _ = vi_tables(im1, im2)
+    # false merges, false splits
```

Comment:
Here you want to update the note to explain that this comment is true *when `image_1` is the test and `image_2` is the reference* (or vice versa, I'm not too sure right now. =P) I might even vote to change the argument names to be more prescriptive (`image_test`, `image_reference`), even though conceptually the VI is symmetric and this wouldn't make any practical difference. But it's better for newcomers to understand.